### ISSUE TITLE
xcolor ignores detected engine and global options

### ISSUE THREAD

**Issue by hmenke** [latex3/xcolor#6]
`xcolor` unconditionally inputs `color.cfg` from `graphics-cfg` which only has very rudimentary engine detection code (detects Lua/pdf/XeTeX, anything else is dvips).  At the same time `xcolor` ignores the global options and any unknown package option, cf. #2.  This results in clobbering of the detected driver with a potentially incorrect one like in the following MWE:

```latex
\documentclass[dvisvgm]{article}
\usepackage{graphics}
\usepackage{xcolor}
\begin{document}
\makeatletter
\show\Gin@driver
\makeatother
\end{document}
```

It is expected that the driver reports `dvisvgm.def` but instead it is

```
> \Gin@driver=macro:
->dvips.def.
```

**Comment by davidcarlisle** [comment]
ah thanks, it sees the global option but just doesn't know it is a driver.

adding `\DeclareOption{dvisvgm}{\def\Gin@driver{dvisvgm.def}}` fixes it. Over the years color and graphics have aquired  several such lines as "current backends" changed, I'll bring the list in xcolor back into sync.

**Comment by u-fischer** [comment]
I checked the list of driver names and get errors for three of them:

~~~~
! LaTeX Error: File `dvipdfm.def' not found.
! LaTeX Error: File `textures.def' not found.
! LaTeX Error: File `vtex.def' not found.
~~~~

**Comment by hmenke** [comment]
The first looks wrong.  The second and third look correct. Textures passed away together with its developer (https://tex.stackexchange.com/a/109051) and VTeX is abandoned (I think). Neither of them supports e-TeX anyway, so they can't even be used with a recent LaTeX kernel.

**Comment by u-fischer** [comment]
This has been resolved in xcolor 2021/10/31 v2.13 and uploaded to ctan.
